src/openfermion/ops/_interaction_tensor.py

```diff
@@ -84,103 +85,114 @@ def two_body_basis_change(two_body_tensor, rotation_matrix):
+        elif 1 in self.n_body_tensors:
+            self.n_qubits = self.n_body_tensors[1].shape[0]
+        else:
+            raise ValueError("Could not determine n_qubits.")
 
     def __getitem__(self, args):
         """Look up matrix element.
 
         Args:
-            Ints giving indices of tensor. Either p,q or p,q,r,s.
+            Ints giving indices of tensor. Should have even length.
 
         Raises:
-            ValueError: args must be of length 2 or 4.
-            ValueError: args must be of length 0, 2 or 4.
+            ValueError: args must be of even length.
         """
-        if len(args) == 4:
-            p, q, r, s = args
-            return self.two_body_tensor[p, q, r, s]
-        elif len(args) == 2:
-            p, q = args
-            return self.one_body_tensor[p, q]
-        elif not len(args):
+        args = tuple(args)
+        if not len(args):
             return self.constant
+        elif len(args) % 2 == 0:
+            return self.n_body_tensors[len(args) // 2][args]
         else:
-            raise ValueError('args must be of length 0, 2, or 4.')
+            raise ValueError('args must be of even length.')
 
     def __setitem__(self, args, value):
         """Set matrix element.
 
         Args:
-            Ints giving indices of tensor. Either p,q or p,q,r,s.
+            Ints giving indices of tensor.
 
         Raises:
-            ValueError: args must be of length 2 or 4.
+            ValueError: args must be of even length.
         """
-        if len(args) == 4:
-            p, q, r, s = args
-            self.two_body_tensor[p, q, r, s] = value
-        elif len(args) == 2:
-            p, q = args
-            self.one_body_tensor[p, q] = value
-        elif not len(args):
+        args = tuple(args)
+        if not len(args):
             self.constant = value
+        elif len(args) % 2 == 0:
+            self.n_body_tensors[len(args) // 2][args] = value
         else:
-            raise ValueError('args must be of length 0, 2, or 4.')
-
-    def __eq__(self, molecular_tensor):
-        constant_diff = abs(molecular_tensor.constant - self.constant)
-        diff = max(constant_diff,
-                   numpy.amax(
-                       numpy.absolute(self.one_body_tensor -
-                                      molecular_tensor.one_body_tensor)),
-                   numpy.amax(
-                       numpy.absolute(self.two_body_tensor -
-                                      molecular_tensor.two_body_tensor)))
+            raise ValueError('args must be of even length.')
+
+    def __eq__(self, other_operator):
+        if self.n_body_tensors.keys() != other_operator.n_body_tensors.keys():
+            return False
+        diff = abs(other_operator.constant - self.constant)
+        for n in self.n_body_tensors.keys():
+            self_tensor = self.n_body_tensors[n]
+            other_tensor = other_operator.n_body_tensors[n]
+            discrepancy = numpy.amax(
```

Comment:
Is there a reason not to use numpy.allclose() with the atol parameter set to EQ_TOLERANCE?

Reply:
The documentation for numpy.allclose says the following:
> If the following equation is element-wise True, then allclose returns True.
>`absolute(a - b) <= (atol + rtol * absolute(b))`
> The above equation is not symmetric in a and b, so that allclose(a, b) might be different from allclose(b, a) in some rare cases.
So it does something slightly different, although it probably doesn't matter.

Reply:
I see.  It looks like the equality check is closer to something like infinity norm of the difference.  Ignore my previous comment.  👍